cvat/apps/annotation/mask.py

```diff
@@ -17,7 +17,7 @@
 }
 
 def dump(file_object, annotations):
-    from zipfile import ZipFile
```

Comment:
@gachiemchiep, I believe we explained wrong what we want to have. In general in the file we should have two dumpers:

```
format_spec = {
    "name": "MASK",
    "dumpers": [
        {
            "display_name": "{name} {format} {version}",
            "format": "ZIP",
            "version": "1.0",
            "handler": "dump_by_class"
        },
        {
            "display_name": "{name} (by instance) {format} {version}",
            "format": "ZIP",
            "version": "1.0",
            "handler": "dump_by_instance"
        },
   ],
    "loaders": [
    ],
}
```

After that instead of one `dump` function let's create two dump functions `dump_by_class` and `dump_by_instance`. Use common `dump` function to implement both and avoid code cloning. Does it make sense?

Reply:
@nmanovic Oh ok. that makes sense now.

Reply:
@nmanovic I update the change.

Reply:
What is the difference between dump_by_instance and dump_by_class? Can we use a common function to implement both?

Reply:
@nmanovic the difference is only which mask will be chosen to write into mask images. I think merging both mask and mask instance into the same dumper method as in the commit before was good enough.

Reply:
> @nmanovic the difference is only which mask will be chosen to write into mask images. I think merging both mask and mask instance into the same dumper method as in the commit before was good enough.

I understand that may be it is annoying to refactor code but it is a really good practice to reduce the number of possible bugs and make the tool a little bit more stable. In this case the only significant difference is how to assign a colour to a pixel (by a label or by a shape index). I will recommend to create `dump(file_object, annotations, get_color)` where get_color is an object which maps `shape.label` and `shape_index` to a color. After that both functions will define `get_color` function and call `dump` with appropriate arguments.

It will help to remove the duplicated code and improve the tool.

Reply:
@nmanovic It's ok. I'll commit the change soon.

Reply:
@nmanovic I understand that you wanted to merge 2 dumper into 1 and add another parameter as method's switch. But after looking at the README.md I don't think I can add another parameter to the dumper method. 

    - dumper/loader handler functions. Each function should have the following signature:
      ```python
      def dump_handler(file_object, annotations):
      ```

Reply:
@gachiemchiep,

You can because `dump` in this case will be an internal function which is used to implement real dumper/loader handlers which will have only the required number of parameters.

Reply:
@nmanovic I committed the change. It should be fine by now

Reply:
@nmanovic Thanks for the advice. I will commit the change soon